manifests/os/redhat.pp

```diff
@@ -8,12 +8,31 @@
   $user              = 'zookeeper',
   $ensure_cron       = true,
   $packages          = ['zookeeper'],
-  $manual_clean      = false
+  $manual_clean      = false,
+  $install_java      = false,
+  $java_package      = undef
 ) {
 
-  # allow installing multiple packages, like zookeeper, zookeeper-bin etc.
-  ensure_resource('package', $packages, {'ensure' => $ensure})
+  validate_bool($install_java)
 
+  # if $install_java, try to make sure a JDK package is installed
+  if ($install_java){
+    if !$java_package {
+      fail { "Java installation is required, but no java package was provided.": }
+    }
+
+    validate_string($java_package)
+
+    # make sure the Java package is only installed once.
+    anchor { 'zookeeper::install::package::begin': }
+    ensure_resource('package', $java_package, {'ensure' => $ensure, 'allow_virtual' => true})
```

Comment:
There's no guarantee that this line will be executed "between" those anchors. Otherwise it looks good, well done :+1: